pkg/rke2/clusterrole.go

```diff
@@ -21,146 +20,33 @@ func setClusterRoles() cmds.StartupHook {
 			<-args.APIServerReady
 			logrus.Info("Applying Cluster Role Bindings")
 
-			cs, err := newClient(args.KubeConfigAdmin, nil)
+			config, err := clientcmd.BuildConfigFromFlags("", args.KubeConfigAdmin)
 			if err != nil {
-				logrus.Fatalf("clusterrole: new k8s client: %s", err.Error())
+				logrus.Fatalf("clusterrole: new k8s client: %v", err)
 			}
 
-			if err := setKubeletAPIServerRoleBinding(ctx, cs); err != nil {
-				logrus.Fatalf("psp: set kubeletAPIServerRoleBinding: %s", err.Error())
-			}
+			stopCh := make(chan struct{})
```

Comment:
As avoidable as Hungarian notation generally is in Go, channels are the one place they're oddly accepted. In this case we should fully accept it.

```suggestion
			stopChan := make(chan struct{})
```